internal/gatewayapi/translator.go

```diff
@@ -136,14 +138,7 @@ func (t *Translator) GetRelevantGateways(gateways []*v1beta1.Gateway) []*Gateway
 			gc := &GatewayContext{
 				Gateway: gateway.DeepCopy(),
 			}
-
-			for _, listener := range gateway.Spec.Listeners {
-				l := gc.GetListenerContext(listener.Name)
-				// Reset conditions and attached route count
-				// since it will be recomputed during translation.
-				l.ResetConditions()
-				l.ResetAttachedRoutes()
-			}
+			gc.ResetListeners()
```

Comment:
* can we delete the `ResetConditions` and `ResetAttachedRoutes` since they are no longer used
* the code within `GetListenerContext` and `ResetListeners` is very similar, can we break up `GetListenerContext` to only perform a Get and use Reset to reset the context ?

Reply:
yea forgot to bring that up but also wanted to ask if we could get rid `ResetConditions` and `ResetAttachedRoutes`

Reply:
yah please delete, also `GetListenerContext` can be deleted too I think
cc @skriss

Reply:
Now that I think about it, `GetListenerContext` is also not used anymore. Wondering if we should remove that method too?

Reply:
sure go for it

Reply:
@arkodg removed unused methods